tools/spec2vec/spec2vec_training.xml

```diff
@@ -0,0 +1,222 @@
+<tool id="spec2vec_training" name="Spec2Vec Model Training" version="@TOOL_VERSION@-@TOOL_DEV_VERSION@+galaxy0" python_template_version="3.5" profile="21.05">
+    <description>Train a Spec2Vec model for mass spectra similarity scoring</description>
+
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="creator"/>
+
+    <requirements>
+        <container type="docker">recetox/spec2vec:@COMMIT_SHA@</container>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        sh ${weights_symlink} &&
+        sh ${spec2vec_python_cli}
+    ]]></command>
+        
+    <configfiles>
+        <configfile name="weights_symlink">
+            ln -fs '${weights_filename}' '${weights_filename}.npy'
+        </configfile>
+        <configfile name="spec2vec_python_cli">
+            python3 ${__tool_directory__}/spec2vec_training_wrapper.py \
+            --spectra_filename '$spectra_filename' \
+            --spectra_fileformat '$spectra_filename.ext' \
+            #if $output_parameters.model_checkpoints.is_true
+            --checkpoints '$output_parameters.model_checkpoints.checkpoints' \
+            #else
+            --epochs $output_parameters.model_checkpoints.epochs \
+            #end if
+            --vector_size $training_parameters.vector_size \
+            --alpha $training_parameters.alpha \
+            --min_alpha $training_parameters.min_alpha \
+            --window $training_parameters.window \
+            --min_count $training_parameters.min_count \
+            --sample $training_parameters.sample \
+            --seed $training_parameters.seed \
+            --sg $training_parameters.sg_param.sg \
+            #if not $training_parameters.sg_param.sg
+            --cbow_mean $training_parameters.sg_param.cbow_mean \
+            #end if
+            --hs $training_parameters.hs_param.hs \
+            #if not $training_parameters.hs_param.hs
+            --negative $training_parameters.hs_param.negative \
+            --ns_exponent $training_parameters.hs_param.ns_exponent \
+            #end if
+            --sorted_vocab $training_parameters.sorted_vocab \
+            --batch_words $training_parameters.batch_words \
+            --shrink_windows $training_parameters.shrink_windows \
+            #if $training_parameters.trim_vocab.max_vocab_size_bool
+            --max_vocab_size $training_parameters.trim_vocab.max_vocab_size \
+            #end if
+            --n_decimals $training_parameters.n_decimals \
+            --n_workers \${GALAXY_SLOTS:-1} \
```

Comment:
_Short answer:_ I think so.

_Long answer:_ When `Slurm` executes a job, it can add `--cpus-per-task <num>` to the call, which can be configured using `Ansible` for the specific tool (using destinations). From what I have [read](https://galaxyproject.org/admin/config/galaxy-slots/), this number should then be available as environment variable `GALAXY_SLOTS`. It is being assigned in [this](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/jobs/runners/util/job_script/CLUSTER_SLOTS_STATEMENT.sh) script.